### PR TITLE
Allow more actions from toolbar to be listed/managed in the customization dialog

### LINKED ISSUE #21923: "Add Circular String" is not deactivatable by Settings > Customization
Author Name: **Thomas Nogatz** (@TomNog)
Original Redmine Issue: [13901](https://issues.qgis.org/issues/13901)

Redmine category:customisation_framework

---
A possibility to deactivate the "Add Circular String" Button would be fine. Cannot find it in Settings > Customization > Toolbars > mDigitizeToolBar.

### PR DESCRIPTION
**Pull request by DelazJ**
This PR fixes #21923 based on the remark at https://github.com/qgis/QGIS/issues/21923#issuecomment-520326904..
Also add more items.

A before (from earlier version) vs after screenshot.
image

## Checklist

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `Fixes #11111` at the bottom of the commit message
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit
- [ ] I have evaluated whether it is appropriate for this PR to be backported, backport requests are left as label or comment

### CODE REVIEW

File: src/app/qgscustomization.cpp [hunk 1]
```diff
@@ -609,7 +610,27 @@ void QgsCustomization::createTreeItemToolbars()
     tbItem->setFlags( Qt::ItemIsEnabled | Qt::ItemIsUserCheckable | Qt::ItemIsSelectable );
     tbItem->setCheckState( 0, Qt::Checked );
 
-    addTreeItemActions( tbItem, tb->actions() );
+    QList<QWidgetAction *> tbWidgetActions = tb->findChildren<QWidgetAction *>( QString(), Qt::FindDirectChildrenOnly );
```
Comment (m-kuhn) [inline src/app/qgscustomization.cpp]: Prefix this line with const

File: src/app/qgscustomization.cpp [hunk 2]
```diff
@@ -742,12 +763,32 @@ void QgsCustomization::updateMainWindow( QMenu *toolBarMenu )
         for ( QAction *action : constActions )
         {
           if ( action->objectName().isEmpty() )
-          {
             continue;
-          }
```
Comment (m-kuhn) [inline src/app/qgscustomization.cpp]: Off topic. We should start to enforce a single style for this via astyle. Half the people add those curly braces for one liners, the other half removes them...

File: src/app/qgscustomization.cpp [hunk 3]
```diff
@@ -742,12 +763,32 @@ void QgsCustomization::updateMainWindow( QMenu *toolBarMenu )
...
+
           visible = mSettings->value( action->objectName(), true ).toBool();
           if ( !visible )
+          {
             tb->removeAction( action );
+            continue;
+          }
+
+          if ( visible && action->metaObject()->className() == QStringLiteral( "QWidgetAction" ) )
```
Comment (m-kuhn) [inline src/app/qgscustomization.cpp]: It will always be visible here, given the above continue...

File: src/app/qgscustomization.cpp [hunk 4]
```diff
@@ -742,12 +763,32 @@ void QgsCustomization::updateMainWindow( QMenu *toolBarMenu )
...
+          {
+            mSettings->beginGroup( action->objectName() );
+            QWidgetAction *widgetAction = qobject_cast<QWidgetAction *>( action );
+            QWidget *widget = widgetAction->defaultWidget();
+            QList<QAction *> childActions = widget->actions();
```
Comment (m-kuhn) [inline src/app/qgscustomization.cpp]: Const

File: src/app/qgscustomization.cpp (hunk 2 above)
Comment (DelazJ) [reply src/app/qgscustomization.cpp]: Indeed. I was surprised to not get feedback from astyle.
What would be the rule? I just remembered an advice of this no-braces style from comments on a PR a while ago and followed it despite no mention is in the Dev guidelines.

File: src/app/qgscustomization.cpp [hunk 5]
```diff
@@ -609,7 +610,27 @@ void QgsCustomization::createTreeItemToolbars()
...
+      QStringList actstrs;
+      actstrs << act->objectName() << act->text();
+      QTreeWidgetItem *item = new QTreeWidgetItem( tbItem, actstrs );
+      item->setIcon( 0, act->icon() );
+      item->setFlags( Qt::ItemIsEnabled | Qt::ItemIsUserCheckable | Qt::ItemIsSelectable );
+      item->setCheckState( 0, Qt::Checked );
+
+      QWidgetAction *widgetAction = qobject_cast<QWidgetAction *>( act );
+      QWidget *widget = widgetAction->defaultWidget();
+      QList<QAction *> childActions = widget->actions();
```
Comment (DelazJ) [inline src/app/qgscustomization.cpp]: const here also?

Comment (m-kuhn) [reply src/app/qgscustomization.cpp]: 1. always when the list is after used "read only"
2. especially when using a range based `for` loop

**Comment by DelazJ** [comment]
I think I addressed all the requests, here.

**Comment by DelazJ** [comment]
Thanks @m-kuhn

**Pull request merged**